crates/transaction-pool/src/validate/eth.rs

```diff
@@ -245,12 +248,12 @@ where
 
         // Checks for gas limit
         let transaction_gas_limit = transaction.gas_limit();
-        if transaction_gas_limit > self.block_gas_limit {
+        if transaction_gas_limit > self.block_gas_limit.load(std::sync::atomic::Ordering::Relaxed) {
```

Comment:
nit: can we convert this into a private `fn max_gas_limit(` then we don't need to have to use load twice